Upgrade to hapi version 20

watson opened the pull request:
Closes #82805

This PR upgrades from hapi v18 to v20 (and related packages).

- [hapi v19 release notes](https://github.com/hapijs/hapi/issues/4017)
- [hapi v20 release notes](https://github.com/hapijs/hapi/issues/4138)
- [hapi issues/PRs labeled `breaking changes`](https://github.com/hapijs/hapi/issues?q=is%3Aclosed+label%3A%22breaking+changes%22+milestone%3A19.0.0)
- [hapi 20.0.3 API docs](https://hapi.dev/api/?v=20.0.3)

## Upgraded modules

| Module                               | Old version      | New version  | Risk         | Changelog        | Breaking Changes
| ------------------------- | -------------- | ------------- | --------- | --------------- | -------------------
| [`@hapi/hapi`]                  | 18.4.1               | 20.0.3             | ❤️            | [Changelog][1] | hapijs/hapi#4017 <br> hapijs/hapi#4138
| [`@hapi/boom`]                | 7.4.11                | 9.1.1               | 💛            | [Changelog][2] | hapijs/boom#251 <br> hapijs/boom#252 <br> hapijs/boom#253 <br> hapijs/boom#262
| [`@hapi/cookie`]              | 10.1.2                | 11.0.2             | 💚            | [Changelog][3] | hapijs/cookie#227
| [`@hapi/good-squeeze`] | 5.2.1                 | 6.0.0              | 💚            | | hapijs/good-squeeze#39
| [`@hapi/h2o2`]                 | 8.3.2                | 9.0.2               | 💛            | [Changelog][5] | hapijs/h2o2#110 <br> hapijs/h2o2#111 <br> hapijs/h2o2#112 <br> hapijs/h2o2#113
| [`@hapi/hoek`]                 | 8.5.1                 | 9.1.0               | ❤️            | [Changelog][6] | hapijs/hoek#343 <br> hapijs/hoek#347
| [`@hapi/inert`]                  | 5.2.2                | 6.0.3               | 💛            | [Changelog][7] | hapijs/inert#137 <br> hapijs/inert#138 <br> hapijs/inert#139 <br> hapijs/inert#140
| [`@hapi/podium`]             | 3.4.3                | 4.1.1                | 💚            | [Changelog][8] | hapijs/podium#57
| [`@hapi/statehood`]        | 6.1.2                 | 7.0.3               | 💚            | [Changelog][9] | hapijs/statehood#66 <br> hapijs/statehood#67
| [`@hapi/vision`]               | 5.4.0                 | 6.0.1               | 💛            | [Changelog][10] | hapijs/vision#190 <br> hapijs/vision#191 <br> hapijs/vision#192
| [`@hapi/wreck`]               | 15.1.0               | 17.1.0               | 💚            | [Changelog][11] | hapijs/wreck#262 <br> hapijs/wreck#263 <br> hapijs/wreck#270

_We're still using our own fork of `@hapi/good` (`@elastic/good`) and `joi` hasn't been upgraded to the newest version._

## Remaining todo's before review

- [x] Address TODOs in code
- [x] Make sure all tests pass

[1]: https://hapi.dev/resources/changelog/
[2]: https://hapi.dev/module/boom/changelog/
[3]: https://hapi.dev/module/cookie/changelog/
[5]: https://hapi.dev/module/h2o2/changelog/
[6]: https://hapi.dev/module/hoek/changelog/
[7]: https://hapi.dev/module/inert/changelog/
[8]: https://hapi.dev/module/podium/changelog/
[9]: https://hapi.dev/module/statehood/changelog/
[10]: https://hapi.dev/module/vision/changelog/
[11]: https://hapi.dev/module/wreck/changelog/
[`@hapi/hapi`]: https://www.npmjs.com/package/@hapi/hapi
[`@hapi/boom`]: https://www.npmjs.com/package/@hapi/boom
[`@hapi/cookie`]: https://www.npmjs.com/package/@hapi/cookie
[`@hapi/good-squeeze`]: https://www.npmjs.com/package/@hapi/good-squeeze
[`@hapi/h2o2`]: https://www.npmjs.com/package/@hapi/h2o2
[`@hapi/hoek`]: https://www.npmjs.com/package/@hapi/hoek
[`@hapi/inert`]: https://www.npmjs.com/package/@hapi/inert
[`@hapi/podium`]: https://www.npmjs.com/package/@hapi/podium
[`@hapi/statehood`]: https://www.npmjs.com/package/@hapi/statehood
[`@hapi/vision`]: https://www.npmjs.com/package/@hapi/vision
[`@hapi/wreck`]: https://www.npmjs.com/package/@hapi/wreck

watson committed:
Bump all hapi packages to the newest version

watson committed:
[@hapi/boom] Implement breaking changes

watson committed:
[typescript][@hapi/boom] Allow output.payload.attributes to exist

watson committed:
[typescript][@hapi/boom] No need to cast output.headers to 'any' anymore

watson committed:
[test][typescript][@hapi/boom] Update Boom mock with new properties

watson committed:
[typescript][@hapi/boom] Cast output.headers to key/value object

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93072/)
* Commit: 11c555a917d7f6676d8c03c6060ecacde74e4b12
* This comment will update when the build is complete

### Failed CI Steps
* [API Integration Tests](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93072/execution/node/353/log)

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93072/","number":"93072","commit":"11c555a917d7f6676d8c03c6060ecacde74e4b12"}]}
PIPELINE-->

watson committed:
Fix route options.validate issue

Hapi handles `{ validate: undefined }` differently from `{}` for route
options.

Having the `validate` key set to `undefined` on the `options` object
resulted in the following error

    Cannot use 'in' operator to search for 'headers' in undefined

This was because the key names were iterated over and their value
expected to be an object.

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93089/)
* Commit: 750c8bc9ebd01529e0b854bf3a2d567f61d9141a
* This comment will update when the build is complete

### Failed CI Steps
* [API Integration Tests](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93089/execution/node/327/log)

### History
* :broken_heart: [Build #93072](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93072/) failed 11c555a917d7f6676d8c03c6060ecacde74e4b12

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93072/","number":"93072","commit":"11c555a917d7f6676d8c03c6060ecacde74e4b12"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93089/","number":"93089","commit":"750c8bc9ebd01529e0b854bf3a2d567f61d9141a"}]}
PIPELINE-->

watson committed:
Revert workaround for hapi pre v20 bug

watson committed:
[hapi] temp fix for payload.multipart breaking change

watson committed:
[test] fix http_server tests

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93540/)
* Commit: 77f5e058c2cca172200e9589349e7722ee879bf9
* This comment will update when the build is complete

### Failed CI Steps
* [Jest Unit Tests](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93540/execution/node/321/log)

### History
* :broken_heart: [Build #93089](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93089/) failed 750c8bc9ebd01529e0b854bf3a2d567f61d9141a
* :broken_heart: [Build #93072](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93072/) failed 11c555a917d7f6676d8c03c6060ecacde74e4b12

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93072/","number":"93072","commit":"11c555a917d7f6676d8c03c6060ecacde74e4b12"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93089/","number":"93089","commit":"750c8bc9ebd01529e0b854bf3a2d567f61d9141a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93540/","number":"93540","commit":"77f5e058c2cca172200e9589349e7722ee879bf9"}]}
PIPELINE-->

watson committed:
Output of 'node scripts/check_published_api_changes --accept'

watson commented on src/core/server/http/http_server.ts:
```diff
@@ -167,66 +167,49 @@ export class HttpServer {
               : undefined,
             timeout: {
-              socket: fakeSocketTimeout,
+              socket: timeout?.idleSocket ?? this.config!.socketTimeout ?? false, // TODO: Do we really want to fall back to NO timeout? Shouldn't we default to the default Node.js timeout of 2 minutes?
```
@restrry I had to fall back to `false` instead of `0` because hapi is validating the value and doesn't allow zero. I'm not sure if `0` previously meant "no timeout", but now `false` means "no timeout", so I think this is what the old code intended.

However, I do think that having no timeout at all is not good practise. But I don't know if we did that on purpose?

restrry replied:
> I do think that having no timeout at all is not good practise.

It seems to be a workaround for introduced by @kobelb in https://github.com/elastic/kibana/pull/73730

https://github.com/elastic/kibana/blob/e116fce08eda66130a2169ece880e26b33cf25e0/src/core/server/http/http_server.ts#L200-L201

As this problem is fixed in hapi v20 it should be okay to fallback to `undefined` @kobelb could you confirm, please?

watson replied:
Thanks for looking into this Mikhail ☺️

I've updated the code to not fall back to `false`, which should mean we just fall back to `undefined` as that should be the value of `this.config.socketTimeout` if it's not set.

We'll see if we can keep it like this based on @kobelb's feedback.

kobelb replied:
> I've updated the code to not fall back to false, which should mean we just fall back to undefined as that should be the value of this.config.socketTimeout if it's not set.

LGTM <3

watson commented on src/core/server/http/http_server.ts:
```diff
@@ -167,66 +167,49 @@ export class HttpServer {
                   output,
                   parse,
                   timeout: timeout?.payload,
+                  multipart: true, // TODO: Figure out what the default value should be
```
@elastic/kibana-core I'm not sure if this is what we want. However, hapi v19 contain the following braking change:

> Change the route `options.payload.multipart` to `false` by default (hapijs/hapi#3920).

The question is if we want to change this to `true` across the board for _all_ routes, or if we want to be more selective about either which route it's applied to, or if we want to use the object-variant instead (see https://hapi.dev/api/?v=20.0.3#-routeoptionspayloadmultipart).

joshdover replied:
I think it would make sense to change the default at one point, but it feels like unnecessary added risk for an upgrade. I think we should do something like that separately.

watson replied:
Sounds good. I've removed the TODO then 👍

restrry replied:
created https://github.com/elastic/kibana/issues/86085

watson commented on src/core/server/http/lifecycle/on_pre_response.ts:
```diff
@@ -142,7 +142,11 @@ export function adoptToHapiOnPreResponseFormat(fn: OnPreResponseHandler, log: Lo
             if (isBoom(response)) {
-              findHeadersIntersection(response.output.headers, result.headers, log);
+              findHeadersIntersection(
+                response.output.headers as { [key: string]: string },
```
The new version of `@hapi/boom` ships with their own types. In these types they changed `output.headers` to `object`, whereas in `@types/hapi__boom` it's defined as `{[index: string]: string}`.

Either we use this hack and add `as { [key: string]: string }` to all places where we expect `ResponseHeaders`, or we update `ResponseHeaders` to allow `object`:

https://github.com/elastic/kibana/blob/008a420f81e10b3c98e2bdd2cca14feef0c61432/src/core/server/http/router/headers.ts#L58-L60

watson commented on src/core/server/http/lifecycle/on_pre_response.ts:
```diff
@@ -142,7 +142,11 @@ export function adoptToHapiOnPreResponseFormat(fn: OnPreResponseHandler, log: Lo
             if (isBoom(response)) {
-              findHeadersIntersection(response.output.headers, result.headers, log);
+              findHeadersIntersection(
+                response.output.headers as { [key: string]: string },
```
@elastic/kibana-core  The new version of `@hapi/boom` ships with their own types. In these types they changed `output.headers` to `object`, whereas in `@types/hapi__boom` it's defined as `{[index: string]: string}`.

Either we use this hack and add `as { [key: string]: string }` to all places where we expect `ResponseHeaders`, or we update `ResponseHeaders` to allow `object`:

https://github.com/elastic/kibana/blob/008a420f81e10b3c98e2bdd2cca14feef0c61432/src/core/server/http/router/headers.ts#L58-L60

joshdover replied:
Personally, I prefer `Record<string, string | string[]>` since it is a more specific type (assuming Hapi isn't parsing those strings, right?)

watson replied:
> Hapi isn't parsing those strings, right?

No, not as far as I know.

watson replied:
I'll keep casting the boom headers manually to `{ [key: string]: string }` then 👍

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93564/)
* Commit: d7466a4855e6145aa1d3b93acdb1ea111d23c9db

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/3440640849b008b0eec8f032f3b2f0c23de717df) | [after](https://github.com/elastic/kibana/commit/d7466a4855e6145aa1d3b93acdb1ea111d23c9db) | diff |
| --- | --- | --- | --- |
| `enterpriseSearch` | 1.7MB | 1.7MB | -211.0B |

#### Distributable file count
| id | [before](https://github.com/elastic/kibana/commit/3440640849b008b0eec8f032f3b2f0c23de717df) | [after](https://github.com/elastic/kibana/commit/d7466a4855e6145aa1d3b93acdb1ea111d23c9db) | diff |
| --- | --- | --- | --- |
| `default` | 47010 | 47608 | +598 |
| `oss` | 27615 | 27453 | -162 |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/3440640849b008b0eec8f032f3b2f0c23de717df) | [after](https://github.com/elastic/kibana/commit/d7466a4855e6145aa1d3b93acdb1ea111d23c9db) | diff |
| --- | --- | --- | --- |
| `upgradeAssistant` | 60.5KB | 60.3KB | -211.0B |

### History
* :broken_heart: [Build #93540](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93540/) failed 77f5e058c2cca172200e9589349e7722ee879bf9
* :broken_heart: [Build #93089](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93089/) failed 750c8bc9ebd01529e0b854bf3a2d567f61d9141a
* :broken_heart: [Build #93072](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93072/) failed 11c555a917d7f6676d8c03c6060ecacde74e4b12

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93072/","number":"93072","commit":"11c555a917d7f6676d8c03c6060ecacde74e4b12"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93089/","number":"93089","commit":"750c8bc9ebd01529e0b854bf3a2d567f61d9141a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93540/","number":"93540","commit":"77f5e058c2cca172200e9589349e7722ee879bf9"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93564/","number":"93564","commit":"d7466a4855e6145aa1d3b93acdb1ea111d23c9db"}]}
PIPELINE-->

watson commented on src/core/server/http/http_server.test.ts:
```diff
@@ -1166,7 +1166,7 @@ describe('timeout options', () => {
         {
           path: '/',
-          validate: { body: schema.any() },
+          validate: false, // TODO: Figure out if we can/should use `{ body: schema.any() }` instead
```
@elastic/kibana-core without this change to the test, the test-server would return:

```js
    {
      statusCode: 400,
      error: 'Bad Request',
      message: '[request body]: expected value of type [any] but got [undefined]'
    }
```

I'm not sure exactly what changed between hapi 18 and 20 that caused this, so I'm not super comfortable with this change to the test. But I can't figure out exactly why it's behaving differently, so I would appreciate another pair of eyes.

joshdover replied:
IIRC, Hapi used to default `request.payload` to an empty object, even if no body was provided. I believe `schema.any` uses `Joi.any` under the hood which does not accept `undefined` by default. If we want to avoid changing this, we could change the value passed into the validator here to default to an empty object: https://github.com/elastic/kibana/blob/master/src/core/server/http/router/request.ts#L144

That said, I actually prefer `undefined` as the default because it maps better to what I would expect from an empty HTTP payload. It allows the application to distinguish between an empty body and an actual `{}` JSON string. If others agree, we could change this test code to `validate: schema.maybe(schema.any())`.

watson replied:
Fixed by wrapping in `schema.maybe` as suggested 👍

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93601/)
* Commit: 49f0180517c3fd33a54672f9322ba5ce2f09c1db

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/b47fd950f34629e4c63f202c6dd8b15246250571) | [after](https://github.com/elastic/kibana/commit/49f0180517c3fd33a54672f9322ba5ce2f09c1db) | diff |
| --- | --- | --- | --- |
| `enterpriseSearch` | 1.7MB | 1.7MB | -211.0B |

#### Distributable file count
| id | [before](https://github.com/elastic/kibana/commit/b47fd950f34629e4c63f202c6dd8b15246250571) | [after](https://github.com/elastic/kibana/commit/49f0180517c3fd33a54672f9322ba5ce2f09c1db) | diff |
| --- | --- | --- | --- |
| `default` | 46931 | 47529 | +598 |
| `oss` | 27534 | 27372 | -162 |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/b47fd950f34629e4c63f202c6dd8b15246250571) | [after](https://github.com/elastic/kibana/commit/49f0180517c3fd33a54672f9322ba5ce2f09c1db) | diff |
| --- | --- | --- | --- |
| `upgradeAssistant` | 60.1KB | 59.9KB | -211.0B |

### History
* :green_heart: [Build #93564](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93564/) succeeded d7466a4855e6145aa1d3b93acdb1ea111d23c9db
* :broken_heart: [Build #93540](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93540/) failed 77f5e058c2cca172200e9589349e7722ee879bf9
* :broken_heart: [Build #93089](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93089/) failed 750c8bc9ebd01529e0b854bf3a2d567f61d9141a
* :broken_heart: [Build #93072](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93072/) failed 11c555a917d7f6676d8c03c6060ecacde74e4b12

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93072/","number":"93072","commit":"11c555a917d7f6676d8c03c6060ecacde74e4b12"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93089/","number":"93089","commit":"750c8bc9ebd01529e0b854bf3a2d567f61d9141a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93540/","number":"93540","commit":"77f5e058c2cca172200e9589349e7722ee879bf9"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93564/","number":"93564","commit":"d7466a4855e6145aa1d3b93acdb1ea111d23c9db"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93601/","number":"93601","commit":"49f0180517c3fd33a54672f9322ba5ce2f09c1db"}]}
PIPELINE-->

watson committed:
Address TODOs

watson commented on src/core/server/http/router/response_adapter.ts:
```diff
@@ -129,21 +129,21 @@ export class HapiResponseAdapter {
 
     const attributes = getErrorAttributes(payload);
     if (attributes) {
+      // @ts-expect-error `attributes` does not exist on the `Payload` type, however it does exist in real life (https://github.com/hapijs/boom/issues/277)
       error.output.payload.attributes = attributes;
```
After a bit of digging (https://github.com/hapijs/boom/issues/277), it turns out that `payload.attributes` probably isn't used correctly here. In fact I can't really figure out why we set this property as it's not really clear that it's used elsewhere.

@mikecote @restrry @pgayvallet three places in this code-base do we set `error.output.payload.attributes`. Do any of you know why/what it's used for?

mikecote replied:
@watson I think this can be cleaned up now. There's two places I can see it still used ([here](https://github.com/elastic/kibana/blob/700225061a3c736aec36ddc12e9669b213894d3b/src/core/server/saved_objects/import/validate_references.ts#L74) and [here](https://github.com/elastic/kibana/blob/b3d97764a0d942f2561ec33266d3d07f267a57a4/src/core/server/saved_objects/export/get_sorted_objects_for_export.ts#L157)) by the saved objects export API.

From what I recall, this was used to indicate what objects caused the API to return 400 (which may not return 400 errors anymore). I'm not sure if the UI got to leverage this but with the new logic it seems this isn't necessary anymore.

mikecote replied:
Also for insight, the usage goes way back to my PR (https://github.com/elastic/kibana/pull/30326) and more specifically, this commit: https://github.com/elastic/kibana/pull/30326/commits/ec662dbc842f194e8a630697910815da5f5443a1.

watson replied:
Great news! Thanks for looking into this 👍 In that case, I'll open a separate PR to get rid of these as it's technically unrelated to this one.

watson replied:
After digging into this, it does seem like we still rely on this. So I'll leave it as is

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/94003/)
* Commit: f7bd546c073452f463abb1e87f214ac92fe2bb6d

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/ab07a003d4b79bc601234612ea0f5b099263af2f) | [after](https://github.com/elastic/kibana/commit/f7bd546c073452f463abb1e87f214ac92fe2bb6d) | diff |
| --- | --- | --- | --- |
| `enterpriseSearch` | 1.7MB | 1.7MB | -211.0B |

#### Distributable file count
| id | [before](https://github.com/elastic/kibana/commit/ab07a003d4b79bc601234612ea0f5b099263af2f) | [after](https://github.com/elastic/kibana/commit/f7bd546c073452f463abb1e87f214ac92fe2bb6d) | diff |
| --- | --- | --- | --- |
| `default` | 47129 | 47727 | +598 |
| `oss` | 27530 | 27368 | -162 |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/ab07a003d4b79bc601234612ea0f5b099263af2f) | [after](https://github.com/elastic/kibana/commit/f7bd546c073452f463abb1e87f214ac92fe2bb6d) | diff |
| --- | --- | --- | --- |
| `upgradeAssistant` | 60.1KB | 59.9KB | -211.0B |

### History
* :green_heart: [Build #93601](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93601/) succeeded 49f0180517c3fd33a54672f9322ba5ce2f09c1db
* :green_heart: [Build #93564](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93564/) succeeded d7466a4855e6145aa1d3b93acdb1ea111d23c9db
* :broken_heart: [Build #93540](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93540/) failed 77f5e058c2cca172200e9589349e7722ee879bf9
* :broken_heart: [Build #93089](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93089/) failed 750c8bc9ebd01529e0b854bf3a2d567f61d9141a
* :broken_heart: [Build #93072](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93072/) failed 11c555a917d7f6676d8c03c6060ecacde74e4b12

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93072/","number":"93072","commit":"11c555a917d7f6676d8c03c6060ecacde74e4b12"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93089/","number":"93089","commit":"750c8bc9ebd01529e0b854bf3a2d567f61d9141a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93540/","number":"93540","commit":"77f5e058c2cca172200e9589349e7722ee879bf9"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93564/","number":"93564","commit":"d7466a4855e6145aa1d3b93acdb1ea111d23c9db"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93601/","number":"93601","commit":"49f0180517c3fd33a54672f9322ba5ce2f09c1db"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/94003/","number":"94003","commit":"f7bd546c073452f463abb1e87f214ac92fe2bb6d"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/94392/)
* Commit: 29b4192b62aa84d8914e1aea008e706d0adeb0ed

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/8ccad311d0455561b9f4a144c1cbecc9a91576b2) | [after](https://github.com/elastic/kibana/commit/29b4192b62aa84d8914e1aea008e706d0adeb0ed) | diff |
| --- | --- | --- | --- |
| `enterpriseSearch` | 1.7MB | 1.7MB | -211.0B |

#### Distributable file count
| id | [before](https://github.com/elastic/kibana/commit/8ccad311d0455561b9f4a144c1cbecc9a91576b2) | [after](https://github.com/elastic/kibana/commit/29b4192b62aa84d8914e1aea008e706d0adeb0ed) | diff |
| --- | --- | --- | --- |
| `default` | 47227 | 47825 | +598 |
| `oss` | 27530 | 27368 | -162 |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/8ccad311d0455561b9f4a144c1cbecc9a91576b2) | [after](https://github.com/elastic/kibana/commit/29b4192b62aa84d8914e1aea008e706d0adeb0ed) | diff |
| --- | --- | --- | --- |
| `upgradeAssistant` | 60.0KB | 59.8KB | -211.0B |

### History
* :green_heart: [Build #94003](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/94003/) succeeded f7bd546c073452f463abb1e87f214ac92fe2bb6d
* :green_heart: [Build #93601](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93601/) succeeded 49f0180517c3fd33a54672f9322ba5ce2f09c1db
* :green_heart: [Build #93564](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93564/) succeeded d7466a4855e6145aa1d3b93acdb1ea111d23c9db
* :broken_heart: [Build #93540](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93540/) failed 77f5e058c2cca172200e9589349e7722ee879bf9
* :broken_heart: [Build #93089](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93089/) failed 750c8bc9ebd01529e0b854bf3a2d567f61d9141a

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93089/","number":"93089","commit":"750c8bc9ebd01529e0b854bf3a2d567f61d9141a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93540/","number":"93540","commit":"77f5e058c2cca172200e9589349e7722ee879bf9"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93564/","number":"93564","commit":"d7466a4855e6145aa1d3b93acdb1ea111d23c9db"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93601/","number":"93601","commit":"49f0180517c3fd33a54672f9322ba5ce2f09c1db"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/94003/","number":"94003","commit":"f7bd546c073452f463abb1e87f214ac92fe2bb6d"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/94392/","number":"94392","commit":"29b4192b62aa84d8914e1aea008e706d0adeb0ed"}]}
PIPELINE-->

chrisronline reviewed (approved):
LGTM for Stack Monitoring!

elasticmachine commented:
Pinging @elastic/ingest-management (Team:Ingest Management)

elasticmachine commented:
Pinging @elastic/apm-ui (Team:apm)

jfsiii reviewed (approved):
👍 for Fleet changes

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/94669/)
* Commit: ff8489f94c659f514954eb9bf81d028ef3fea93f

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/23cd2670e1649d1eb149e6cf51628ef20675dfa8) | [after](https://github.com/elastic/kibana/commit/ff8489f94c659f514954eb9bf81d028ef3fea93f) | diff |
| --- | --- | --- | --- |
| `enterpriseSearch` | 1.7MB | 1.7MB | -211.0B |

#### Distributable file count
| id | [before](https://github.com/elastic/kibana/commit/23cd2670e1649d1eb149e6cf51628ef20675dfa8) | [after](https://github.com/elastic/kibana/commit/ff8489f94c659f514954eb9bf81d028ef3fea93f) | diff |
| --- | --- | --- | --- |
| `default` | 47242 | 47840 | +598 |
| `oss` | 27540 | 27378 | -162 |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/23cd2670e1649d1eb149e6cf51628ef20675dfa8) | [after](https://github.com/elastic/kibana/commit/ff8489f94c659f514954eb9bf81d028ef3fea93f) | diff |
| --- | --- | --- | --- |
| `upgradeAssistant` | 60.0KB | 59.8KB | -211.0B |

### History
* :green_heart: [Build #94392](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/94392/) succeeded 29b4192b62aa84d8914e1aea008e706d0adeb0ed
* :green_heart: [Build #94003](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/94003/) succeeded f7bd546c073452f463abb1e87f214ac92fe2bb6d
* :green_heart: [Build #93601](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93601/) succeeded 49f0180517c3fd33a54672f9322ba5ce2f09c1db
* :green_heart: [Build #93564](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93564/) succeeded d7466a4855e6145aa1d3b93acdb1ea111d23c9db
* :broken_heart: [Build #93540](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93540/) failed 77f5e058c2cca172200e9589349e7722ee879bf9

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93540/","number":"93540","commit":"77f5e058c2cca172200e9589349e7722ee879bf9"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93564/","number":"93564","commit":"d7466a4855e6145aa1d3b93acdb1ea111d23c9db"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93601/","number":"93601","commit":"49f0180517c3fd33a54672f9322ba5ce2f09c1db"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/94003/","number":"94003","commit":"f7bd546c073452f463abb1e87f214ac92fe2bb6d"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/94392/","number":"94392","commit":"29b4192b62aa84d8914e1aea008e706d0adeb0ed"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/94669/","number":"94669","commit":"ff8489f94c659f514954eb9bf81d028ef3fea93f"}]}
PIPELINE-->

smith reviewed (approved):
APM changes look good.

flash1293 reviewed (approved):
Kibana app changes LGTM, type changes only. Didn't test locally

darnautov reviewed (approved):
ML and Transform changes LGTM

watson committed:
Merge branch 'master' into bump-hapi

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/95091/)
* Commit: f90a66c500e059325eeeab5ab58c12a50ee841aa

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/5be169a4fc9e59de5be129449c974783f6a24a75) | [after](https://github.com/elastic/kibana/commit/f90a66c500e059325eeeab5ab58c12a50ee841aa) | diff |
| --- | --- | --- | --- |
| `enterpriseSearch` | 1.7MB | 1.7MB | -211.0B |

#### Distributable file count
| id | [before](https://github.com/elastic/kibana/commit/5be169a4fc9e59de5be129449c974783f6a24a75) | [after](https://github.com/elastic/kibana/commit/f90a66c500e059325eeeab5ab58c12a50ee841aa) | diff |
| --- | --- | --- | --- |
| `default` | 47288 | 47886 | +598 |
| `oss` | 27865 | 27703 | -162 |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/5be169a4fc9e59de5be129449c974783f6a24a75) | [after](https://github.com/elastic/kibana/commit/f90a66c500e059325eeeab5ab58c12a50ee841aa) | diff |
| --- | --- | --- | --- |
| `upgradeAssistant` | 60.1KB | 59.9KB | -211.0B |

### History
* :green_heart: [Build #94669](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/94669/) succeeded ff8489f94c659f514954eb9bf81d028ef3fea93f
* :green_heart: [Build #94392](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/94392/) succeeded 29b4192b62aa84d8914e1aea008e706d0adeb0ed
* :green_heart: [Build #94003](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/94003/) succeeded f7bd546c073452f463abb1e87f214ac92fe2bb6d
* :green_heart: [Build #93601](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93601/) succeeded 49f0180517c3fd33a54672f9322ba5ce2f09c1db
* :green_heart: [Build #93564](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93564/) succeeded d7466a4855e6145aa1d3b93acdb1ea111d23c9db

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93564/","number":"93564","commit":"d7466a4855e6145aa1d3b93acdb1ea111d23c9db"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93601/","number":"93601","commit":"49f0180517c3fd33a54672f9322ba5ce2f09c1db"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/94003/","number":"94003","commit":"f7bd546c073452f463abb1e87f214ac92fe2bb6d"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/94392/","number":"94392","commit":"29b4192b62aa84d8914e1aea008e706d0adeb0ed"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/94669/","number":"94669","commit":"ff8489f94c659f514954eb9bf81d028ef3fea93f"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/95091/","number":"95091","commit":"f90a66c500e059325eeeab5ab58c12a50ee841aa"}]}
PIPELINE-->

legrego reviewed (commented):
I think we can remove this patch now that Hapi supports `SameSite=None` natively:

https://github.com/elastic/kibana/blob/700225061a3c736aec36ddc12e9669b213894d3b/src%2Fcore%2Fserver%2Fhttp%2Fcookie_session_storage.ts#L162-L177

restrry commented:
> I think we can remove this patch now that Hapi supports SameSite=None natively:

@legrego I'm going to take care of it as soon as the PR is merged

watson committed:
Merge branch 'master' into bump-hapi

watson committed:
Bump @hapi/boom to v9.1.1

This fixes a TypeScript error which means we can get rid of a few
ts-expect-error comments.

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/95347/)
* Commit: 9be257bc82448c28f4796b166e8b2d2572843a5f
* This comment will update when the build is complete

### Failed CI Steps
* [Check Types](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/95347/execution/node/369/log)

### History
* :green_heart: [Build #95091](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/95091/) succeeded f90a66c500e059325eeeab5ab58c12a50ee841aa
* :green_heart: [Build #94669](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/94669/) succeeded ff8489f94c659f514954eb9bf81d028ef3fea93f
* :green_heart: [Build #94392](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/94392/) succeeded 29b4192b62aa84d8914e1aea008e706d0adeb0ed
* :green_heart: [Build #94003](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/94003/) succeeded f7bd546c073452f463abb1e87f214ac92fe2bb6d
* :green_heart: [Build #93601](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93601/) succeeded 49f0180517c3fd33a54672f9322ba5ce2f09c1db

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/93601/","number":"93601","commit":"49f0180517c3fd33a54672f9322ba5ce2f09c1db"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/94003/","number":"94003","commit":"f7bd546c073452f463abb1e87f214ac92fe2bb6d"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/94392/","number":"94392","commit":"29b4192b62aa84d8914e1aea008e706d0adeb0ed"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/94669/","number":"94669","commit":"ff8489f94c659f514954eb9bf81d028ef3fea93f"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/95091/","number":"95091","commit":"f90a66c500e059325eeeab5ab58c12a50ee841aa"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/95347/","number":"95347","commit":"9be257bc82448c28f4796b166e8b2d2572843a5f"}]}
PIPELINE-->

legrego reviewed (approved):
LGTM on green CI!

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/95360/)
* Commit: d77b07990e763c957c142f6ca1f5fc994498dcd9

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/011849faea9ae37883262e304ae692cd37565d44) | [after](https://github.com/elastic/kibana/commit/d77b07990e763c957c142f6ca1f5fc994498dcd9) | diff |
| --- | --- | --- | --- |
| `enterpriseSearch` | 1.7MB | 1.7MB | -211.0B |

#### Distributable file count
| id | [before](https://github.com/elastic/kibana/commit/011849faea9ae37883262e304ae692cd37565d44) | [after](https://github.com/elastic/kibana/commit/d77b07990e763c957c142f6ca1f5fc994498dcd9) | diff |
| --- | --- | --- | --- |
| `default` | 47287 | 47885 | +598 |
| `oss` | 27863 | 27701 | -162 |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/011849faea9ae37883262e304ae692cd37565d44) | [after](https://github.com/elastic/kibana/commit/d77b07990e763c957c142f6ca1f5fc994498dcd9) | diff |
| --- | --- | --- | --- |
| `upgradeAssistant` | 60.1KB | 59.9KB | -211.0B |

### History
* :broken_heart: [Build #95347](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/95347/) failed 9be257bc82448c28f4796b166e8b2d2572843a5f
* :green_heart: [Build #95091](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/95091/) succeeded f90a66c500e059325eeeab5ab58c12a50ee841aa
* :green_heart: [Build #94669](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/94669/) succeeded ff8489f94c659f514954eb9bf81d028ef3fea93f
* :green_heart: [Build #94392](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/94392/) succeeded 29b4192b62aa84d8914e1aea008e706d0adeb0ed
* :green_heart: [Build #94003](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/94003/) succeeded f7bd546c073452f463abb1e87f214ac92fe2bb6d

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/94003/","number":"94003","commit":"f7bd546c073452f463abb1e87f214ac92fe2bb6d"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/94392/","number":"94392","commit":"29b4192b62aa84d8914e1aea008e706d0adeb0ed"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/94669/","number":"94669","commit":"ff8489f94c659f514954eb9bf81d028ef3fea93f"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/95091/","number":"95091","commit":"f90a66c500e059325eeeab5ab58c12a50ee841aa"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/95347/","number":"95347","commit":"9be257bc82448c28f4796b166e8b2d2572843a5f"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/95360/","number":"95360","commit":"d77b07990e763c957c142f6ca1f5fc994498dcd9"}]}
PIPELINE-->

rylnd reviewed (approved):
Security Solution changes LGTM!
